### PR TITLE
Make WCSAxes.set_axes_off() work

### LINKED ISSUE #8919: .set_axis_off() doesn't work with WCSAxes
`ax.set_axis_off()` should turn off all the axis annotations, but it doesn't do anything on a `WCSAxes`:

```python
import matplotlib.pyplot as plt

from astropy.wcs import WCS
from astropy.io import fits
from astropy.utils.data import get_pkg_data_filename

filename = get_pkg_data_filename('galactic_center/gc_msx_e.fits')

hdu = fits.open(filename)[0]
wcs = WCS(hdu.header)

fig = plt.figure()
ax1 = plt.subplot(2, 1, 1, projection=wcs)
ax2 = plt.subplot(2, 1, 2)
for ax in [ax1, ax2]:
    ax.set_axis_off()
    ax.plot([0, 1], [0, 1])

plt.show()
```

Figure_1

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #8919 I'm not sure what the best way is to test this - maybe a figure test?

### CODE REVIEW

**Comment by Cadair** [comment]
yeah I would think a figure test would be the only way?

**Comment by dstansby** [comment]
I've added an image test, but I don't have access to docker, would someone else mind generating the figure test image?

**Comment by bsipocz** [comment]
@Cadair - as you're already doing it for your PR(s), could you please take of the figure generating from here, too, and update them in one go?

**Comment by Cadair** [comment]
yeah, no worries.

File: astropy/visualization/wcsaxes/tests/test_images.py [hunk 1]
```diff
@@ -58,6 +58,16 @@ def test_image_plot(self):
...
         return fig
 
+    @pytest.mark.remote_data(source='astropy')
+    @pytest.mark.mpl_image_compare(baseline_dir=IMAGE_REFERENCE_DIR,
+                                   tolerance=0, style={})
+    def test_axes_off(self):
+        # Test for turning the axes off
+        fig = plt.figure(figsize=(6, 6))
+        ax = fig.add_axes([0.1, 0.1, 0.8, 0.8], projection=WCS(self.msx_header))
+        ax.set_axis_off()
```
Comment (astrofrog) [inline astropy/visualization/wcsaxes/tests/test_images.py]: Can you add plotting of an image to make sure there is something in the figure and clipped to the axes?

File: astropy/visualization/wcsaxes/tests/test_images.py [hunk 2]
```diff
@@ -58,6 +58,16 @@ def test_image_plot(self):
         ax.coords[0].set_ticks([-0.30, 0., 0.20] * u.degree, size=5, width=1)
         return fig
 
+    @pytest.mark.remote_data(source='astropy')
+    @pytest.mark.mpl_image_compare(baseline_dir=IMAGE_REFERENCE_DIR,
+                                   tolerance=0, style={})
+    def test_axes_off(self):
+        # Test for turning the axes off
+        fig = plt.figure(figsize=(6, 6))
```
Comment (astrofrog) [inline astropy/visualization/wcsaxes/tests/test_images.py]: Can you make this smaller? (e.g. 3, 3)

**Comment by astrofrog** [comment]
In the interest of getting this into 4.0, I've updated the test along the lines of my suggestions above - hope you don't mind @dstansby! I've also added the reference images to data.astropy.org.

**Comment by bsipocz** [comment]
Thanks @dstansby!

**Comment by dstansby** [comment]
Thanks for generating the test image @astrofrog!

**Pull request merged**